Server/src/main/java/octoteam/tahiti/server/pipeline/SessionExpireHandler.java

```diff
@@ -2,21 +2,36 @@
 
 import io.netty.channel.ChannelHandler;
 import io.netty.channel.ChannelHandlerContext;
-import octoteam.tahiti.protocol.SocketMessageProtos;
+import octoteam.tahiti.protocol.SocketMessageProtos.Message;
+import octoteam.tahiti.server.TahitiServer;
+import octoteam.tahiti.server.event.RateLimitExceededEvent;
 
 @ChannelHandler.Sharable
 public class SessionExpireHandler extends InboundMessageHandler {
 
     @Override
-    public void channelRead0(ChannelHandlerContext ctx, SocketMessageProtos.Message msg) {
+    public void channelRead0(ChannelHandlerContext ctx, Message msg) {
         ctx.fireChannelRead(msg);
     }
 
     @Override
     public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
 
-        // TODO
-
+        if (evt instanceof RateLimitExceededEvent) {
+            RateLimitExceededEvent event = (RateLimitExceededEvent)evt;
+            if (event.getName().equals(RateLimitExceededEvent.NAME_PER_SESSION)) {
+                Message.Builder resp = Message.newBuilder()
+                        .setDirection(Message.DirectionCode.RESPONSE)
+                        .setService(Message.ServiceCode.SESSION_EXPIRED_EVENT);
+                ctx.channel().writeAndFlush(resp.build());
+                setSession(ctx, null);
+            } else {
+                Message.Builder resp = Message.newBuilder()
+                        .setDirection(Message.DirectionCode.RESPONSE)
+                        .setStatus(Message.StatusCode.LIMIT_EXCEEDED);
+                ctx.channel().writeAndFlush(resp.build());
```

Comment:
SessionExpireHandler should not send LIMIT_EXCEEDED and LIMIT_EXCEEDED is already sent in [RateLimitHandler](https://github.com/SummerWish/Tahiti/blob/master/Server/src/main/java/octoteam/tahiti/server/pipeline/RateLimitHandler.java#L35)